parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,197 @@
+		substitutionSchedule.setClasses(getAllClasses());
+		substitutionSchedule.setTeachers(getAllTeachers());
+		final JSONArray changes = getChanges();
+		final JSONArray courses = getCourses();
+		final JSONArray teachers = getTeachers();
+
+		parseLegionBoard(substitutionSchedule, changes, courses, teachers);
+		return substitutionSchedule;
+	}
+
+	/**
+	 * Returns authentication key as shown
+	 * <a href="https://gitlab.com/legionboard/heart/blob/master/doc/README.md">in the documentation</a>.
+	 */
+	private String getAuthenticationKey(Credential credential) {
+		final UserPasswordCredential userPasswordCredential = (UserPasswordCredential) credential;
+		final String username = userPasswordCredential.getUsername();
+		final String password = userPasswordCredential.getPassword();
+		return DigestUtils.sha256Hex(username.toLowerCase() + "//" + password);
+	}
+
+	/**
+	 * Returns a JSONArray with all changes from now to in one week.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/changes/list.md">List changes</a>
+	 */
+	private JSONArray getChanges() throws IOException, JSONException {
+		// Date (or alias of date) when the changes start
+		final String startBy = "now";
+		// Date (or alias of date) when the changes end
+		final String endBy = "i1w";
+
+		final String url = api + "/changes?startBy=" + startBy + "&endBy=" + endBy + "&k=" + getAuthenticationKey(getCredential());
+		return getJSONArray(url);
+	}
+
+	/**
+	 * Returns a JSONArray with all courses.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/courses/list.md">List courses</a>
+	 */
+	private JSONArray getCourses() throws IOException, JSONException {
+		final String url = api + "/courses?k=" + getAuthenticationKey(getCredential());
+		try {
+			return getJSONArray(url);
+		} catch (HttpResponseException e) {
+			// TODO: Implement
+			final boolean is404 = true;
+			if (is404) {
+				return null;
+			}
+			throw e;
+		}
+	}
+
+	/**
+	 * Returns a JSONArray with all teachers.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/teachers/list.md">List teachers</a>
+	 */
+	private JSONArray getTeachers() throws IOException, JSONException {
+		final String url = api + "/teachers?k=" + getAuthenticationKey(getCredential());
+		try {
+			return getJSONArray(url);
+		} catch (HttpResponseException e) {
+			// TODO: Implement
+			final boolean is404 = true;
+			if (is404) {
+				return null;
+			}
+			throw e;
+		}
+	}
+
+	private JSONArray getJSONArray(String url) throws IOException, JSONException, HttpResponseException {
+		return new JSONArray(httpGet(url, "UTF-8"));
+	}
+
+	private void parseLegionBoard(SubstitutionSchedule substitutionSchedule, JSONArray changes, JSONArray courses, JSONArray teachers) throws IOException, JSONException {
+		if (changes == null) {
+			throw new IOException("LegionBoard has no changes.");
```

Comment:
Kann man in der App die Änderungen nach Datum filtern? Heart kann nämlich theoretisch alle Änderungen, die jemals erstellt wurden, anzeigen. Wenn das andere Systeme nicht können (ich kenn mich da nicht aus), dann kann das die App wohl eher nicht, oder?

On July 23, 2016 1:41:49 PM GMT+02:00, Johan von Forstner <notifications@github.com> wrote:
>> +			// TODO: Implement
>> +			final boolean is404 = true;
>> +			if (is404) {
>> +				return null;
>> +			}
>> +			throw e;
>> +		}
>> +	}
>> +
>> +	private JSONArray getJSONArray(String url) throws IOException,
>JSONException, HttpResponseException {
>> +		return new JSONArray(httpGet(url, "UTF-8"));
>> +	}
>> +
>> +	private void parseLegionBoard(SubstitutionSchedule
>substitutionSchedule, JSONArray changes, JSONArray courses, JSONArray
>teachers) throws IOException, JSONException {
>> +		if (changes == null) {
>> +			throw new IOException("LegionBoard has no changes.");
>
>Das ist je nach Schule unterschiedlich. Bei den meisten gibt es nur nur
>heute und morgen, bei manchen sogar nur heute. Bei manchen (mit dem
>Untis-Infostundenplan zum Beispiel) aber auch ~1 Woche. Wenn bei Eye
>eine Woche der Standard ist, würde ich das hier auch so machen.
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/johan12345/substitution-schedule-parser/pull/6/files/c18972380dbbcfa4d07b02a7366a6277712bc391#r71971404